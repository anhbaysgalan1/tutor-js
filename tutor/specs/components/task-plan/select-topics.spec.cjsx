{React, SnapShot} = require '../helpers/component-testing'

SelectTopics = require '../../../src/components/task-plan/select-topics'
{default: Courses} = require '../../../src/models/courses-map'

{TaskPlanActions, TaskPlanStore} = require '../../../src/flux/task-plan'
{TocStore, TocActions} = require '../../../src/flux/toc'

{ExtendBasePlan, PlanRenderHelper} = require '../helpers/task-plan'
{HomeworkPlan} = require '../../../src/components/task-plan/homework'

COURSE_ID = '1'
ECOSYSTEM_ID = '3'
COURSE = require '../../../api/user/courses/1.json'
HOMEWORK = require '../../../api/plans/2.json'
READINGS  = require '../../../api/ecosystems/3/readings.json'

helper = (model) -> PlanRenderHelper(model, HomeworkPlan)

describe 'Select Topics', ->

  beforeEach ->
    TocActions.reset()
    TaskPlanActions.reset()
    Courses.bootstrap([COURSE], { clear: true })
    TocActions.loaded(READINGS, ECOSYSTEM_ID)
    TaskPlanActions.loaded(HOMEWORK, HOMEWORK.id)
    @props =
      ecosystemId: HOMEWORK.ecosystem_id
      planId: HOMEWORK.id
      courseId: COURSE_ID
      onSectionChange: jest.fn()
      hide: jest.fn()
      selected: []
      header: 'Section Chooser Header'
      cancel: jest.fn()

  it 'renders using ecosystem of task', ->
    wrapper = mount(<SelectTopics {...@props} />)
    expect(wrapper.find(".select-chapters[data-ecosystem-id=\"#{@props.ecosystemId}\"]")).to.have.length(1)
    undefined

  it 'matches snapshot', ->
    expect(SnapShot.create(<SelectTopics {...@props} />).toJSON()).toMatchSnapshot()
    undefined
