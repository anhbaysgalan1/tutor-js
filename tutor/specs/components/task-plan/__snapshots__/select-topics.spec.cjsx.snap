// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Topics matches snapshot 1`] = `
<div
  className="loadable is-loading"
>
  Loading... 
</div>
`;
